torch/_higher_order_ops/templated_attention.py

```diff
@@ -220,3 +249,403 @@ def templated_attention_fake_tensor_mode(
             batch_size, num_heads, seq_len_q, dtype=torch.float32
         )
         return torch.empty_like(query, memory_format=torch.contiguous_format), logsumexp
+
+
+# ---------------------------- Autograd Implementation ----------------------------
+def create_fw_bw_graph(score_mod, index_values, other_buffers):
+    # See Note:[HOP create fw_bw graph]
+
+    # All of these imports need to be here in order to avoid circular dependencies
+    from torch._dispatch.python import suspend_functionalization
+    from torch._functorch.aot_autograd import AOTConfig, create_joint, from_fun
+
+    from torch._subclasses.functional_tensor import (
+        disable_functional_mode,
+        FunctionalTensor,
+    )
+    from torch.fx.experimental.proxy_tensor import disable_proxy_modes_tracing
+
+    dummy_aot_config = AOTConfig(
+        fw_compiler=None,  # type: ignore[arg-type]
+        bw_compiler=None,  # type: ignore[arg-type]
+        partition_fn=None,  # type: ignore[arg-type]
+        decompositions={},
+        num_params_buffers=0,
+        aot_id=0,
+        keep_inference_input_mutations=False,
+    )
+    with suspend_functionalization(), disable_functional_mode():
+        with disable_proxy_modes_tracing():
+
+            def _from_fun(t):
+                if isinstance(t, torch.Tensor):
+                    if t.dtype != torch.bool:
+                        # New empty is causing some bogus values to be attempted to index
+                        return torch.zeros(
```

Comment:
DONT LAND

during the course of compilation, this is called multiple times, with empty_strided we are attempting to acces the other_buffes with a junk index and this causes an error, I feel like we dont really care about the strides since all the inputs are scalars. 

But this is also applied to the other_buffers so maybe we should use empty_strided there?

or is there an zero_strided?

Reply:
sadly that breaks things  for leaf var reasons, maybe worth trying out of place